### ISSUE TITLE
[Question]: Is it possible to not load content based on response content-length header

### ISSUE THREAD

**Issue by cyberfuhrer** [microsoft/playwright-python#1385]
### Your question

I try to intercept the response in route to get the content length header and then block it. 
Example https://playwright.dev/python/docs/next/network#modify-responses return AssertionError: First argument must be either URL string or Request

**Comment by cyberfuhrer** [comment]
```
from playwright.sync_api import sync_playwright

def handle_route(route):
	response = page.request.fetch(route.request)

playwright = sync_playwright().start()
browser = playwright.chromium.launch(headless=True)
context = browser.new_context()
page = context.new_page()
page.route("**/*", handle_route)
page.goto(url='https://example.com/', timeout=10000)
```
```
Exception in callback SyncBase._sync.<locals>.<lambda>(<Task finishe... or Request')>) at /usr/local/lib/python3.7/dist-packages/playwright/_impl/_sync_base.py:84
handle: <Handle SyncBase._sync.<locals>.<lambda>(<Task finishe... or Request')>) at /usr/local/lib/python3.7/dist-packages/playwright/_impl/_sync_base.py:84>
Traceback (most recent call last):
  File "/usr/lib/python3.7/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.7/dist-packages/playwright/_impl/_sync_base.py", line 84, in <lambda>
    task.add_done_callback(lambda _: g_self.switch())
  File "/usr/local/lib/python3.7/dist-packages/playwright/_impl/_page.py", line 198, in <lambda>
    from_channel(params["route"]), from_channel(params["request"])
  File "/usr/local/lib/python3.7/dist-packages/playwright/_impl/_page.py", line 242, in _on_route
    handler_entry.handle(route, request)
  File "/usr/local/lib/python3.7/dist-packages/playwright/_impl/_helper.py", line 225, in handle
    )(route, request)
  File "/usr/local/lib/python3.7/dist-packages/playwright/_impl/_impl_to_api_mapping.py", line 116, in wrapper_func
    *list(map(lambda a: self.from_maybe_impl(a), args))[:arg_count]
  File "plw_block.py", line 4, in handle_route
    response = page.request.fetch(route.request)
  File "/usr/local/lib/python3.7/dist-packages/playwright/sync_api/_generated.py", line 14393, in fetch
    ignoreHTTPSErrors=ignore_https_errors,
  File "/usr/local/lib/python3.7/dist-packages/playwright/_impl/_sync_base.py", line 88, in _sync
    return task.result()
  File "/usr/local/lib/python3.7/dist-packages/playwright/_impl/_fetch.py", line 248, in fetch
    ), "First argument must be either URL string or Request"
AssertionError: First argument must be either URL string or Request
```
Playwright version 1.22.0
Thanks

**Comment by rwoll** [comment]
Oops—it looks like our docs are wrong on the next branch! 🤦 Sorry about that, and thanks for helping surface it.

Just opened https://github.com/microsoft/playwright/pull/15162 to fix it. You can change your code above:

```
def handle_route(route):
	response = page.request.fetch(route.request)
```

to be:

```
def handle_route(route, _request):
	response = page.request.fetch(route.request)
```

Python is expecting to be able to pass 2 arguments when calling your handler.

Does this resolve your issue?

**Comment by rwoll** [comment]
It looks like we had a bug in 1.22; it will be fixed in 1.23 via this [commit](https://github.com/microsoft/playwright-python/issues/1312). 1.23 will be released in the next few days. Sorry for the inconvenience!

I'm going to close this issue, but please re-file after 1.23 if you encounter any issues.
